autogen/agentchat/contrib/vectordb/pgvectordb.py

```diff
@@ -0,0 +1,710 @@
+import os
+import re
+from typing import Callable, List
+
+import numpy as np
+from sentence_transformers import SentenceTransformer
+
+from .base import Document, ItemID, QueryResults, VectorDB
+from .utils import get_logger
+
+try:
+    import pgvector
+    from pgvector.psycopg import register_vector
+except ImportError:
+    raise ImportError("Please install pgvector: `pip install pgvector`")
+
+try:
+    import psycopg
+except ImportError:
+    raise ImportError("Please install pgvector: `pip install psycopg`")
+
+PGVECTOR_MAX_BATCH_SIZE = os.environ.get("PGVECTOR_MAX_BATCH_SIZE", 40000)
+logger = get_logger(__name__)
+
+
+class Collection:
```

Comment:
This is nice! Thank you for the example for creating collection for all sql style databases.